### PR TITLE
[8.0] fix(wms): correctly log the pilot job reference during the matching process

### PR DESCRIPTION
**Pull request by aldbr**
Closes https://github.com/DIRACGrid/DIRAC/issues/7884

BEGINRELEASENOTES
*WorkloadManagement
FIX: correctly log the pilot job reference during the matching process
ENDRELEASENOTES

### CODE REVIEW

**Comment by aldbr** [comment]
https://github.com/DIRACGrid/DIRAC/actions/runs/12144642670/job/33864474014?pr=7924#step:10:935

```python
--- Logging error ---
Traceback (most recent call last):
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/logging/__init__.py", line 1113, in emit
    stream.write(msg + self.terminator)
ValueError: I/O operation on closed file.
Call stack:
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/weakref.py", line 666, in _exitfunc
    f()
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/weakref.py", line 590, in __call__
    return info.func(*info.args, **(info.kwargs or {}))
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/site-packages/DIRAC/Core/Utilities/DictCache.py", line 247, in _purgeAll
    deleteFunction(cache[cKey]["value"])
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/site-packages/DIRAC/WorkloadManagementSystem/DB/TaskQueueDB.py", line 1001, in __deleteTQIfEmpty
    result = self.deleteTaskQueueIfEmpty(tqId, tqOwnerDN, tqOwnerGroup)
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/site-packages/DIRAC/WorkloadManagementSystem/DB/TaskQueueDB.py", line 1027, in deleteTaskQueueIfEmpty
    retVal = self._query(sqlCmd, conn=connObj)
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/site-packages/DIRAC/Core/Utilities/MySQL.py", line 717, in _query
    self.log.debug(f"_query: {self._safeCmd(cmd)}")
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/site-packages/DIRAC/FrameworkSystem/private/standardLogging/Logging.py", line 352, in debug
    return self._createLogRecord(LogLevels.DEBUG, sMsg, sVarMsg)
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/site-packages/DIRAC/FrameworkSystem/private/standardLogging/Logging.py", line 417, in _createLogRecord
    self._logger.log(level, "%s", sMsg, exc_info=exc_info, extra=extra)
Message: '%s'
Arguments: ('_query: SELECT TQId FROM `tq_TaskQueues` WHERE Enabled >= 1 AND `tq_TaskQueues`.TQId = 26 AND `tq_TaskQueues`.TQId not in ( SELECT DISTINCT TQId from `tq_Jobs` )',)
```

**Comment by aldbr** [comment]
Update: ok funny, this is not a new bug: https://github.com/DIRACGrid/DIRAC/actions/runs/12140642296/job/33850877559

I am not sure to fully understand, but it looks like it is due to the `weakref`. When we are in `__deleteTQIfEmpty`, it's like the streams have already been cleaned up, and the logger, even if it's still alive, cannot emit the messages anymore.

May be we should not try to log/print messages in methods used by `weakref`? But then we take the risk of missing some errors

File: src/DIRAC/WorkloadManagementSystem/Service/MatcherHandler.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def export_requestJob(self, resourceDescription):
 
         resourceDescription["Setup"] = self.serviceInfoDict["clientSetup"]
         credDict = self.getRemoteCredentials()
-        pilotRef = resourceDescription.get("PilotReference", "Unknown")
+        pilotRef = resourceDescription.get("PilotReference")
```
Comment (fstagni) [inline src/DIRAC/WorkloadManagementSystem/Service/MatcherHandler.py]: This was not harming.

File: src/DIRAC/FrameworkSystem/Utilities/LoggingContext.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+""" Logging context module"""
```
Comment (fstagni) [inline src/DIRAC/FrameworkSystem/Utilities/LoggingContext.py]: All the logging code is in `DIRAC.FrameworkSystem.private.standardLogging` right? You can add this code in one of the existing module (here it feels a bit "lost").

File: src/DIRAC/WorkloadManagementSystem/Service/MatcherHandler.py (hunk 1 above)
Comment (aldbr) [reply src/DIRAC/WorkloadManagementSystem/Service/MatcherHandler.py]: Then, what's the point of having the following piece of code here?

https://github.com/DIRACGrid/DIRAC/pull/7924/files#diff-6e5904ade77b6cd702bb41532101e8027ba0ebc3a3430c98332efd246df5af5eR55-R56

File: src/DIRAC/FrameworkSystem/Utilities/LoggingContext.py (hunk 2 above)
Comment (aldbr) [reply src/DIRAC/FrameworkSystem/Utilities/LoggingContext.py]: Indeed, but then it becomes weird to import a package coming from `<something>.private.<something_else>`.

May be the best solution would be to add it somewhere in `DIRAC.FrameworkSystem.private.standardLogging` and then call it through `DIRAC.FrameworkSystem.Client.Logger`.

File: src/DIRAC/WorkloadManagementSystem/Service/MatcherHandler.py (hunk 1 above)
Comment (fstagni) [reply src/DIRAC/WorkloadManagementSystem/Service/MatcherHandler.py]: It was probably to handle cases of old pilots without a pilotRef.

Comment (aldbr) [reply src/DIRAC/WorkloadManagementSystem/Service/MatcherHandler.py]: Thanks for the clarification, I reverted my change.

File: src/DIRAC/FrameworkSystem/Utilities/LoggingContext.py (hunk 2 above)
Comment (aldbr) [reply src/DIRAC/FrameworkSystem/Utilities/LoggingContext.py]: I applied the changes

**Review by chrisburr — APPROVED** [review]

**Pull request merged**